store/store.go

```diff
@@ -352,6 +352,22 @@ func refreshMacaroon(user *auth.UserState) error {
 	return nil
 }
 
+// authenticateDevice will add the store expected Macaroon X-Device-Authorization header for device
+func (s *Store) authenticateDevice(r *http.Request) {
+	if s.authContext != nil {
+		device, err := s.authContext.Device()
+		if err != nil {
+			logger.Debugf("cannot get device from state: %v", err)
+			return
+		}
+		if device != nil && device.SessionMacaroon != "" {
+			var buf bytes.Buffer
+			fmt.Fprintf(&buf, `Macaroon root="%s"`, device.SessionMacaroon)
```

Comment:
This feels like it can be abbreviate to just:
```
r.Header.Set("X-Device-Authorization", fmt.Sprintf(`Macaroon root="%s"`, device.SessionMacaroon))
```